### ISSUE TITLE
Layout Header missing accessibility role

### ISSUE THREAD

**Issue by khalibloo** [ant-design/ant-design#43718]
### Reproduction link

[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/3f96dc)

### Steps to reproduce

1. Add `<Layout.Header />` inside `<Layout />`

### What is expected?

The `<header />` element in the DOM should be a landmark banner. It should have the `banner` role in the browser's accessibility inspector.

### What is actually happening?

The `<header />` is placed inside a `<section />` which nullifies its landmark status according to https://developer.mozilla.org/en-US/docs/Web/HTML/Element/header#accessibility

| Environment | Info |
| --- | --- |
| antd | 5.7.2 |
| React | 18.2.0 |
| System | Windows 10 |
| Browser | Firefox 115 |

---
`<Layout />` should output a `<div />` instead of a `<section />`.
